jsonschema/_utils.py

```diff
@@ -181,25 +217,27 @@ def uniq(container):
     """
     Check if all of a container's elements are unique.
 
-    Successively tries first to rely that the elements are hashable, then
-    falls back on them being sortable, and finally falls back on brute
-    force.
+    Successively tries first to rely that the elements are being sortable
+    and finally falls back on brute force.
     """
-
     try:
-        return len(set(unbool(i) for i in container)) == len(container)
-    except TypeError:
-        try:
-            sort = sorted(unbool(i) for i in container)
-            sliced = itertools.islice(sort, 1, None)
-            for i, j in zip(sort, sliced):
-                if i == j:
+        sort = sorted(unbool(i) for i in container)
+        sliced = itertools.islice(sort, 1, None)
+
+        for i, j in zip(sort, sliced):
+            return not list_equal(list(i), list(j))
+
+    except (NotImplementedError, TypeError):
+        seen = []
+        for e in container:
+            e = unbool(e)
+
+            for i in seen:
+                if isinstance(i, dict) and isinstance(e, dict):
```

Comment:
This line too I'm suspicious of if it still contains `isinstance(, dict)` -- it'll fail for non-dict mappings then, so we need to exercise that case as well.

Let me know if that makes sense.

Reply:
This has been removed and it makes use of the `equal` function instead.